benchmarks/benchmarks.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require "benchmark/ips"
+
+require "memery"
+require "memo_wise"
+require "memoist"
+require "memoized"
+require "memoizer"
+
+BenchmarkGem = Struct.new(:klass, :inheritance_method, :memoization_method)
+
+# We alphabetize this list for easier readability, but shuffle the list before
+# using it to minimize the chance that our benchmarks are affected by ordering.
+BENCHMARK_GEMS = [
+  BenchmarkGem.new(Memery, :include, :memoize),
+  BenchmarkGem.new(MemoWise, :prepend, :memo_wise),
+  BenchmarkGem.new(Memoist, :extend, :memoize),
+  BenchmarkGem.new(Memoized, :include, :memoize),
+  BenchmarkGem.new(Memoizer, :include, :memoize)
+].shuffle
+
+# Use metaprogramming to ensure that each class is created in exactly the
+# the same way.
+BENCHMARK_GEMS.each do |benchmark_gem|
+  Object.const_set("#{benchmark_gem.klass}Example", Class.new do
+    send(benchmark_gem.inheritance_method, benchmark_gem.klass)
+
+    def no_args
+      100
+    end
+    send(benchmark_gem.memoization_method, :no_args)
+
+    def positional_args(a, b)
+      200 + a + b
+    end
+    send(benchmark_gem.memoization_method, :positional_args)
+
+    def keyword_args(a:, b:)
+      300 + a + b
+    end
+    send(benchmark_gem.memoization_method, :keyword_args)
+  end)
+end
+
+# We benchmark different cases separately, to ensure that slow performance in
+# one method or code path isn't hidden by fast performance in another.
+Benchmark.ips do |x|
+  BENCHMARK_GEMS.each do |benchmark_gem|
+    example_class = Object.const_get("#{benchmark_gem.klass}Example")
+    x.report("#{benchmark_gem.klass}#no_args") do |n_times|
+      n_times.times do
+        instance = example_class.new
+        10_000.times { instance.no_args }
+      end
+    end
+  end
+
+  x.compare!
```

Comment:
Initial results—not great:

```
Comparison:
    Memoized#no_args:      882.6 i/s
     Memoist#no_args:      421.9 i/s - 2.09x  (± 0.00) slower
    Memoizer#no_args:      334.9 i/s - 2.64x  (± 0.00) slower
      Memery#no_args:       85.6 i/s - 10.31x  (± 0.00) slower
    MemoWise#no_args:       75.1 i/s - 11.75x  (± 0.00) slower
```

Reply:
Not for this PR, but I wonder if it's worth branching on the no args case?